src/icq/system.rs

```diff
@@ -0,0 +1,98 @@
+use super::protocol;
+use crate::messages::{
+    AccountHandle, AccountInfo, FdSender, ICQSystemHandle, PurpleMessage, SystemMessage,
+};
+use async_std::sync::{channel, Receiver};
+use log;
+
+const CHANNEL_CAPACITY: usize = 1024;
+
+pub fn spawn() -> ICQSystemHandle {
+    let (input_rx, input_tx) = os_pipe::pipe().unwrap();
+    let (system_tx, system_rx) = channel(CHANNEL_CAPACITY);
+    let (purple_tx, purple_rx) = channel(CHANNEL_CAPACITY);
+
+    let fd_sender = FdSender::new(input_tx, system_tx);
+
+    log::debug!("Starting async thread.");
+    std::thread::spawn(move || run(fd_sender, purple_rx));
+
+    ICQSystemHandle {
+        input_rx,
+        rx: system_rx,
+        tx: purple_tx,
+    }
+}
+
+pub fn run(tx: FdSender<SystemMessage>, rx: Receiver<PurpleMessage>) {
+    log::info!("Starting ICQ");
+    let mut system = ICQSystem::new(tx, rx);
+    async_std::task::block_on(system.run());
+}
+
+pub struct ICQSystem {
+    tx: FdSender<SystemMessage>,
+    rx: Receiver<PurpleMessage>,
+}
+
+impl ICQSystem {
+    fn new(tx: FdSender<SystemMessage>, rx: Receiver<PurpleMessage>) -> Self {
+        Self { tx, rx }
+    }
+
+    async fn run(&mut self) {
+        loop {
+            let purple_message = match self.rx.recv().await {
+                Ok(r) => r,
+                Err(error) => {
+                    log::error!("Failed to receive message: {:?}", error);
+                    break;
+                }
+            };
+            log::info!("Message: {:?}", purple_message);
+            match purple_message {
+                PurpleMessage::Login(account_info) => self.login(account_info).await,
+            }
+            self.ping().await
```

Comment:
Tu ping le serveur nonstop?

Reply:
Pour test les messages, removed in following PR